File: android/app/src/main/java/com/locationsharing/app/domain/repository/LocationRepository.kt
package com.locationsharing.app.domain.repository

import com.locationsharing.app.data.models.Location
import kotlinx.coroutines.flow.Flow

interface LocationRepository {
    suspend fun updateLocation(location: Location): Result<Unit>
    suspend fun getCurrentLocation(): Location?
    fun getLocationUpdates(): Flow<Location>
    suspend fun getFriendLocations(friendIds: List<String>): Result<Map<String, Location>>
    suspend fun startLocationSharing(): Result<Unit>
    suspend fun stopLocationSharing(): Result<Unit>
}

File: ios/LocationSharingApp/LocationSharingApp/Domain/Repositories/LocationRepository.swift
import Foundation
import Combine

protocol LocationRepository {
    func updateLocation(_ location: Location) async -> Result<Void, Error>
    func getCurrentLocation() async -> Location?
    func getLocationUpdates() -> AnyPublisher<Location, Never>
    func getFriendLocations(friendIds: [String]) async -> Result<[String: Location], Error>
    func startLocationSharing() async -> Result<Void, Error>
    func stopLocationSharing() async -> Result<Void, Error>
}